caput/tests/test_mpiarray.py

```diff
@@ -394,7 +397,69 @@ def test_global_setslice(self):
 
         assert (darr == local_array).all()
 
+    def test_outer_ufunc(self):
+        rank = mpiutil.rank
+        size = mpiutil.size
+
+        dist_arr = mpiarray.MPIArray((size, 4), axis=0)
+        dist_arr[:] = rank
+
+        dist_arr_scalar = mpiarray.MPIArray((size, 4), axis=0)
+        dist_arr_scalar[:] = rank * 2
+
+        dist_arr_add = dist_arr + dist_arr
+
+        # Check that you can add two numpy arrays,
+        # if they are distributed along the same axes
+        # Check that you can multiple a numpy array against a scalar
+        assert (dist_arr_add == dist_arr_scalar).all()
```

Comment:
Also, note that `.all()` in an MPIArray reduction over all axes, and we hadn't clearly worked out the semantics for that yet.

Seeing how it works with `.all()` maybe pushes it in the direction of you should get one value per rank.